src/scripts/validation.js

```diff
@@ -0,0 +1,78 @@
+function showInputError(form, input, errorMessage, config) {
+  const errorElement = form.querySelector(`#${input.id}-error`);
+
+  errorElement.textContent = errorMessage;
+  errorElement.classList.add(config.errorClass);
+
+  input.classList.add(config.inputErrorClass);
+}
+
+function hideInputError(form, input, config) {
+  const errorElement = form.querySelector(`#${input.id}-error`);
+
+  errorElement.textContent = "";
+  errorElement.classList.remove(config.errorClass);
+
+  input.classList.remove(config.inputErrorClass);
+}
+
+function checkInputValidity(form, input, config) {
+  if (input.validity.valid) {
+    hideInputError(form, input, config);
+  } else {
+    showInputError(form, input, input.validationMessage, config);
+  }
+}
+
+export function checkFormValidity(form, config) {
+  const inputs = Array.from(form.querySelectorAll(config.inputSelector));
+  const button = form.querySelector(config.submitButtonSelector);
+
+  toggleButtonState(inputs, button);
+
+  inputs.forEach((input) => {
+    checkInputValidity(form, input, config);
+    toggleButtonState(inputs, button);
```

Comment:
This is not needed here because you have it in line 31


Only `hideInputError` should be called here

```js
  inputs.forEach((input) => {
    hideInputError(form, input, config);
  });

```